include/aws/mqtt/client.h

```diff
@@ -245,6 +249,7 @@ int aws_mqtt_client_connection_connect(
     const struct aws_byte_cursor *client_id,
     bool clean_session,
     uint16_t keep_alive_time,
+    uint32_t request_timeout_ms,
```

Comment:
While we're messing with this, should we put these (clean_session, keep_alive_time, request_timeout_ms) into a ```mqtt_connection_options``` struct so the API doesn't change after this, should we need a new config value?